src/LessMsi.Cli/Program.cs

```diff
@@ -124,10 +186,17 @@ private static void PrintProgress(IAsyncResult result)
             Console.WriteLine(string.Format("{0}/{1}\t{2}", progress.FilesExtractedSoFar + 1, progress.TotalFileCount, progress.CurrentFileName));
         }
 
-        private static void EnsureFileRooted(ref string sFileName)
-		{
-			if (!Path.IsPathRooted(sFileName))
-				sFileName = Path.Combine(Directory.GetCurrentDirectory(), sFileName);
-		}
-	}
+        private static void EnsureFileRooted(ref string sFileName, eExtractionMode extractionMode = eExtractionMode.RegularExtraction)
+        {
+            if (!Path.IsPathRooted(sFileName))
+            {
+                sFileName = Path.Combine(Directory.GetCurrentDirectory(), sFileName);
+            }
+
+            if (isExtractionModeFlat(extractionMode))

```

Comment:
Are you sure this is what you want in the case of !Path.IsPathRooted (the if above)? Now that this method has become more complicated it is probably worth a unit test with the various combinations of extractionMode and with scenarios where sFileName is rooted and when it is not rooted.

Reply:
How about I leave the `EnsureFileRooted` method as it was originally and extract my added logic into a separate method?
This way the logic in `EnsureFileRooted` method stays the same and I can call the newly created method whenever I need without involving `EnsureFileRooted` method.